spec.html

```diff
@@ -8468,9 +8468,9 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-functionallocate" aoid="FunctionAllocate">
-      <h1>FunctionAllocate ( _functionPrototype_, _functionKind_ )</h1>
-      <p>The abstract operation FunctionAllocate requires the two arguments _functionPrototype_ and _functionKind_. FunctionAllocate performs the following steps:</p>
+    <emu-clause id="sec-ordinaryfunctioncreate" aoid="OrdinaryFunctionCreate" oldids="sec-functionallocate,sec-functioninitialize">
+      <h1>OrdinaryFunctionCreate ( _functionPrototype_, _ParameterList_, _Body_, _functionKind_, _kind_, _Scope_ )</h1>
```

Comment:
It's unfortunate there's both a `_kind_` and a `_functionKind_`. AFAICT the `~Method~` case of `_kind_` doesn't do anything, and `_kind_` is only used to determine the `this` scope lookup mode. I would prefer to pass that directly, i.e. a new `_thisMode_` that's `~LexicalThis~` or `~NonLexicalThis~` and have the non-lexical case checked for strictness inside this AO.

@jmdyck WDYT?

Reply:
Ah, currently `_kind_` is still used to determine the allocation kind for non-constructibles. It's all rather circuitous logic in there. Do you plan to layer this on top of #1546, after which my suggestion above makes more sense?

Reply:
> Do you plan to layer this on top of #1546?

Well, if #1546 is accepted, then I'll resolve the merge conflicts, if that's what you mean.

Reply:
> I would prefer to pass that directly, i.e. a new `_thisMode_` that's `~LexicalThis~` or `~NonLexicalThis~`

Done! (modulo naming)

> and have the non-lexical case checked for strictness inside this AO.

I wasn't sure what you meant by this.